ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/execution/ListSourceExecutor.java

```diff
@@ -191,14 +200,42 @@ private static SourceDescriptionWithWarnings describeSource(
       ), statementText);
     }
 
+    List<RunningQuery> sourceQueries = getQueries(ksqlEngine,
+        q -> q.getSourceNames().contains(dataSource.getName()));
+    List<RunningQuery> sinkQueries = getQueries(ksqlEngine,
+        q -> q.getSinkName().equals(dataSource.getName()));
+
     Optional<org.apache.kafka.clients.admin.TopicDescription> topicDescription = Optional.empty();
+    Optional<ConsumerGroupDescription> consumerGroupDescription = Optional.empty();
+    Map<TopicPartition, OffsetAndMetadata> topicAndConsumerOffsets = new LinkedHashMap<>();
+    Map<TopicPartition, ListOffsetsResultInfo> topicAndStartOffsets = new LinkedHashMap<>();
+    Map<TopicPartition, ListOffsetsResultInfo> topicAndEndOffsets = new LinkedHashMap<>();
     final List<KsqlWarning> warnings = new LinkedList<>();
     if (extended) {
       try {
         topicDescription = Optional.of(
             serviceContext.getTopicClient().describeTopic(dataSource.getKafkaTopicName())
         );
-      } catch (final KafkaException | KafkaResponseGetFailedException e) {
+        String serviceId = "default"; //FIXME not sure how to get this
+        if (sourceQueries.isEmpty()){
+          consumerGroupDescription = Optional.empty();
+        } else {
+          String queryId = sourceQueries.get(0).getId().toString();
```

Comment:
Would be possible to get more than one source query?

Reply:
yes, this is in the case of a JOIN you could possibly have more than one source query

Reply:
I'm wondering how it will have more than one source when there is only one topic:
```
        final TopicDescription topicDescription = serviceContext.getTopicClient()
            .describeTopic(dataSource.getKafkaTopicName());
```

Thinking out loud: When a query is joining 2 streams, is acting on top of the streams, not a topic itself, therefore no consumer group.